source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1349,6 +1465,26 @@ clear the server's connection pool if and only if the error is
 error messages to detect "not master" and "node is recovering"`_, and `other
 transient errors`_ and `Why close connections when a node is shutting down?`_.)
 
+Reducing race conditions in error handling
+``````````````````````````````````````````
+
+Clients MUST NOT handle errors that have stale generation numbers. That is,
+when handling a network or command error, if the error's generation number
+is not equal to the current pool's generation number then the error MUST be
+ignored.
+
+Clients MUST NOT handle errors that have stale toplogyVersions. That is,
+a client MUST NOT handle a command error unless one or more of the following
+conditions are true:
+
+#. The current ServerDescription's topologyVersion is null.
```

Comment:
Here's a third place where the comparison algorithm is defined.  I think this needs to be DRY'd up